utils/statediff/src/main.rs

```diff
@@ -1,68 +1,63 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
+use clap::Parser;
 use directories::ProjectDirs;
 use forest_db::db_engine::{db_path, open_db, DbConfig};
-use structopt::StructOpt;
 
 use cid::Cid;
 use forest_statediff::print_state_diff;
 
-/// Examine the state delta
-#[derive(StructOpt)]
-pub struct ChainCommand {
-    /// The previous CID state root
-    pre: Cid,
-    /// The post CID state root
-    post: Cid,
-    /// The name of the chain
-    #[structopt(short, long, default_value = "mainnet")]
-    chain: String,
-    /// The depth at which IPLD links are resolved
-    #[structopt(short, long)]
-    depth: Option<u64>,
-}
-
-impl ChainCommand {
+impl crate::Subcommand {
     pub fn run(&self) -> anyhow::Result<()> {
-        let dir = ProjectDirs::from("com", "ChainSafe", "Forest")
-            .ok_or(anyhow::Error::msg("no such path"))?;
-        let chain_path = dir.data_dir().join(&self.chain);
-        let blockstore = open_db(&db_path(&chain_path), &DbConfig::default())?;
+        match self {
+            Subcommand::ChainCommand {
+                pre,
+                post,
+                chain,
+                depth,
+            } => {
+                let dir = ProjectDirs::from("com", "ChainSafe", "Forest")
+                    .ok_or(anyhow::Error::msg("no such path"))?;
+                let chain_path = dir.data_dir().join(chain);
+                let blockstore = open_db(&db_path(&chain_path), &DbConfig::default())?;
 
-        if let Err(err) = print_state_diff(&blockstore, &self.pre, &self.post, self.depth) {
-            eprintln!("Failed to print state diff: {err}");
+                if let Err(err) = print_state_diff(&blockstore, pre, post, *depth) {
+                    eprintln!("Failed to print state diff: {err}");
+                }
+                Ok(())
+            }
         }
-        Ok(())
     }
 }
 
-/// statediff binary sub-commands available.
-#[derive(StructOpt)]
-#[structopt(setting = structopt::clap::AppSettings::VersionlessSubcommands)]
-enum Subcommand {
-    #[structopt(name = "chain")]
-    Chain(ChainCommand),
-}
-
 /// CLI structure generated when interacting with the statediff tool
-#[derive(StructOpt)]
-#[structopt(
-    name = env!("CARGO_PKG_NAME"),
-    version = env!("CARGO_PKG_VERSION"),
-    about = env!("CARGO_PKG_DESCRIPTION"),
-    author = env!("CARGO_PKG_AUTHORS")
-)]
```

Comment:
Why did you delete it? The binaries lost, amongst others, versioning. Do you agree it's incorrect?

```
more_forest on  tyshko5/issue-2365-to-clap via 🦀 v1.69.0-nightly
❯ target/debug/forest_statediff --version
error: unexpected argument '--version' found

Usage: forest_statediff <COMMAND>

For more information, try '--help'.
```
vs
```
more_forest on  main via 🦀 v1.69.0-nightly took 40s
❯ target/debug/forest_statediff --version
forest_statediff 0.6.0

```

Reply:
Reverted to previous behaviour

Reply:
@tyshko5 we still miss the author in the help.

structopt
```
❯ target/debug/forest_statediff --help
forest_statediff 0.6.0
ChainSafe Systems <info@chainsafe.io>
statediff binary sub-commands available

USAGE:
    forest_statediff <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    chain    Examine the state delta
    help     Prints this message or the help of the given subcommand(s)

```
clap (this branch)
```
❯ target/debug/forest_statediff --help

Usage: forest_statediff <COMMAND>

Commands:
  chain  
  help   Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version

```

Could you make it, so we don't lose any of that information? Otherwise, passing the author seems pointless if we don't use it anywhere. [This](https://docs.rs/clap/latest/clap/builder/struct.Command.html#method.help_template) may be needed.

Reply:
The thing you've mentioned can only be used only through `Builder` pattern, and we are using `Derive`. Theoretically there are the same attributes [here](https://docs.rs/clap/latest/clap/_derive/index.html#command-attributes) , but they just  don't work. So I'd just delete those fields @LesnyRumcajs

Reply:
Finally works as before @LesnyRumcajs